### ISSUE TITLE
Surge quit unexpectedly after running for around 1 day

### ISSUE THREAD

**Issue by Chewie9999** [rule110-io/surge#53]
**Describe the bug**

I left Surge running since I tried to replicate the bug #52 and when I checked it a few minutes ago, it had quit.
See attached log file [log.zip](https://github.com/rule110-io/surge/files/5837338/log.zip)
`xxx.xxx.xxx.xxx` is my desktop's public IP address

**To Reproduce**
?

**Expected behavior**
Surge should not quit unless asked :)

**Screenshots**
None, it had quit when I found it.

**Desktop (please complete the following information):**
 - OS: openSUSE Tumbleweed 20210110
 - Version: v0.3.2-beta

**Additional context**
There were some "go" errors on my console, but sadly I did not capture them. Hopefully the log file is enough.
I have started surge again to see if it does the same thing during the next day or so.
If it does, I will save the console output too.

**Comment by MutsiMutsi** [comment]
Hi @Chewie9999 thanks for reporting, it looks like the logs caught the cause here, will investigate!

**Comment by MutsiMutsi** [comment]
Do you know whether the machine on which Surge runs has general issues with internet connectivity, or perhaps you are in a region which is prone to very high latency or network outage?

The logs describe connectivity issues at multiple levels, which would point towards your machine as opposed to remote causes. However this should never crash surge of course, we will try to resolve that. 

Could you kindly confirm the above could hold true?

**Comment by Chewie9999** [comment]
It might have been a low memory situation as later that day, everything was crawling to a halt. I tried to find out what could have caused this, but have not been able to figure it out. Normally I have ~17Gb used out of 32
One thing I did notice is that in htop, virtual memory ("VIRT") for surge was 100G. After a reboot Surge has now been running for 2 days, and VIRT is "only" 86.5G.

Network connectivity should not have been a problem. My other devices on the same internet connection were working fine.
